src/libostree/ostree-bootloader-zipl.c

```diff
@@ -80,6 +83,203 @@ _ostree_bootloader_zipl_write_config (OstreeBootloader  *bootloader,
   return TRUE;
 }
 
+
+
+static gboolean
+_ostree_secure_execution_get_keys (GArray **keys, GError **error)
+{
+  const char *keyspath = "/etc";
+  g_autoptr(GFile) path          = g_file_new_for_path (keyspath);
+  g_autoptr(GFileEnumerator) etc = g_file_enumerate_children (path, G_FILE_ATTRIBUTE_STANDARD_NAME,
+                                                              G_FILE_QUERY_INFO_NONE, NULL, error);
+  if (*error)
+    {
+      sd_journal_print(LOG_ERR, "s390x SE: looking for SE keys: %s", (*error)->message);
+      return FALSE;
+    }
+  *keys = g_array_new (FALSE, FALSE, sizeof(GString*));
```

Comment:
`GString` is like `StringBuilder` in Java; it's less common in C to pass it around.  Much more common here is using `GPtrArray` with `NUL` terminated strings, i.e.:
`*keys = g_ptr_array_new_with_free_func (g_free);`